extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueBridgeHandler.java

```diff
@@ -519,4 +519,13 @@ private boolean isEqual(State state1, State state2) {
 
         return configStatusMessages;
     }
+
+    @Override
+    public String getDeviceId() {
+        try {
+            return bridge.getFullConfig().getConfig().getMACAddress().replaceAll(":", "").toLowerCase();
```

Comment:
The [HueThingHandlerFactory](https://github.com/eclipse/smarthome/blob/0d2b3b8c88cdc5cfa4efe3b44584f67712e93e6b/extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/HueThingHandlerFactory.java#L68) reads it from the configuration, the [HueBridgeDiscoveryParticipant](https://github.com/eclipse/smarthome/blob/0d2b3b8c88cdc5cfa4efe3b44584f67712e93e6b/extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/discovery/HueBridgeDiscoveryParticipant.java#L46) gets if from the device directly, the [HueBridgeNupnpDiscovery](https://github.com/eclipse/smarthome/blob/7a32b35d08035ce81aa5831796740ff02c035c56/extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/discovery/HueBridgeNupnpDiscovery.java#L83) applies some other magic - why yet a fourth way of creatively calculating the serial number? Can't you also read it from the handler configuration?